scipy/stats/_stats_py.py

```diff
@@ -9244,41 +9244,47 @@ def combine_pvalues(pvalues, method='fisher', weights=None):
     .. [8] https://en.wikipedia.org/wiki/Extensions_of_Fisher%27s_method
 
     """
-    if pvalues.size == 0:
+    xp = array_namespace(pvalues)
+    pvalues = xp.asarray(pvalues)
+    if xp_size(pvalues) == 0:
         NaN = _get_nan(pvalues)
         return SignificanceResult(NaN, NaN)
 
     if method == 'fisher':
-        statistic = -2 * np.sum(np.log(pvalues))
-        chi2 = _SimpleChi2(2 * len(pvalues))
+        statistic = -2 * xp.sum(xp.log(pvalues))
+        chi2 = _SimpleChi2(xp.asarray(2 * xp_size(pvalues), dtype=pvalues.dtype))
         pval = _get_pvalue(statistic, chi2, alternative='greater',
-                           symmetric=False, xp=np)
+                           symmetric=False, xp=xp)
     elif method == 'pearson':
-        statistic = 2 * np.sum(np.log1p(-pvalues))
-        chi2 = _SimpleChi2(2 * len(pvalues))
+        statistic = 2 * xp.sum(xp.log1p(-pvalues))
+        chi2 = _SimpleChi2(xp.asarray(2 * xp_size(pvalues), dtype=pvalues.dtype))
         pval = _get_pvalue(-statistic, chi2, alternative='less',
-                           symmetric=False, xp=np)
+                           symmetric=False, xp=xp)
     elif method == 'mudholkar_george':
-        normalizing_factor = np.sqrt(3/len(pvalues))/np.pi
-        statistic = -np.sum(np.log(pvalues)) + np.sum(np.log1p(-pvalues))
-        nu = 5 * len(pvalues) + 4
-        approx_factor = np.sqrt(nu / (nu - 2))
-        pval = distributions.t.sf(statistic * normalizing_factor
-                                  * approx_factor, nu)
+        normalizing_factor = math.sqrt(3/xp_size(pvalues))/xp.pi
+        statistic = -xp.sum(xp.log(pvalues)) + xp.sum(xp.log1p(-pvalues))
+        nu = xp.asarray(5 * xp_size(pvalues) + 4, dtype=pvalues.dtype)
+        approx_factor = math.sqrt(nu / (nu - 2))
+        t = _SimpleStudentT(nu)
+        pval = _get_pvalue(statistic * normalizing_factor * approx_factor, t,
+                           alternative="greater", symmetric=False, xp=xp)
```

Comment:
```suggestion
                           alternative="greater", xp=xp)
```
The t distribution is symmetric, so this isn't needed.